docs/kb/logging.md

```diff
@@ -192,6 +188,113 @@ If the replicating process loses its connection to the logging process, you can
 Currently, only a single replicating process can subscribe to the primary process. If another kdb+ process attempts to replicate from the primary, the previous replicating process will no longer receive updates. If you need multiple replicating processes, you might like to consider [kdb+tick](../learn/startingkdb/tick.md).
 
 
+
+## Replaying log files
+
+
+Streaming-execute over a file is used (for example in kdb+tick) to replay a log file in a memory-efficient manner.
+
+:fontawesome-solid-book-open:
+[`-11!` streaming execute](../basics/internal.md#-11-streaming-execute)
+
+A logfile is just a list of lists, and each list is read in turn and evaluated by [`.z.ps`](../ref/dotz.md#zps-set) (which defaults to [`value`](../ref/value.md)).
+
+Here, for demonstration purposes, we manually create a logfile, and play it back through `-11!`. This is functionally equivalent to doing ``value each get `:logfile`` but uses far less memory.
+
+```q
+q)`:logfile.2013.12.03 set () / create a new,empty log file
+`:logfile.2013.12.03
+q)h:hopen `:logfile.2013.12.03 / open it
+q)h enlist(`f;`a;10) / append a record
+3i
+q)h enlist(`f;`b;20) / append a record
+3i
+q)hclose h / close the file
+q)/Define the function that is referenced in those records
+q)f:{0N!(x;y)}
+q)-11!`:logfile.2013.12.03 / playback the logfile
+(`a;10)
+(`b;20)
+2
+q)/ DO NOT DO THIS ON LARGE LOGFILES!!!!
+q)/This is the whole purpose of -11!x.
+q)value each get `:logfile.2013.12.03
+(`a;10)
+(`b;20)
+`a 10
+`b 20
+```
+
+If successful, the number of chunks executed is returned. 
+
+If the end of the file is corrupt a `badtail` error is signalled, which may be partially [recovered](#replay-from-corrupt-logs).
+
+In the event that the log file references an undefined function, the function name is signalled as an error. This can be confusing if the missing function name is `upd`, as it does not reflect the same situation as the license expiry `upd` error. e.g.
```

Comment:
```suggestion
In the event that the log file references an undefined function, the function name is signalled as an error. This can be confusing if the missing function name is `upd`, as it does not reflect the same situation as the license expiry `upd` error. For example:.
```